authorization/role/service/role_management_service_blackbox_test.go

```diff
@@ -196,3 +200,71 @@ func (s *roleManagementServiceBlackboxTest) TestGetIdentityRoleByResourceAndRole
 	require.NoError(t, err)
 	require.Equal(t, 0, len(identityRoles))
 }
+
+func (s *roleManagementServiceBlackboxTest) TestAssignRoleOK() {
+	g := s.DBTestSuite.NewTestGraph()
+	newSpace := g.CreateSpace(g.ID("myspace"))
+
+	var usersToBeAsserted []uuid.UUID
+	for i := 0; i < 10; i++ {
+		userToBeAssigned := g.CreateUser(g.ID("adminname"))
+		err := s.repo.Assign(context.Background(), userToBeAssigned.Identity().ID, newSpace.SpaceID(), authorization.AdminRole)
+		require.Nil(s.T(), err)
+		usersToBeAsserted = append(usersToBeAsserted, userToBeAssigned.Identity().ID)
+	}
+
+	// noise
+	for i := 0; i < 10; i++ {
+		err := s.repo.Assign(context.Background(), g.CreateUser(g.ID("adminname")).Identity().ID, g.CreateSpace(g.ID("myspace")).SpaceID(), authorization.AdminRole)
+		require.Nil(s.T(), err)
+	}
+
+	assignedRoles, err := s.repo.ListByResourceAndRoleName(context.Background(), newSpace.SpaceID(), authorization.AdminRole)
+	require.Nil(s.T(), err)
+	require.Len(s.T(), assignedRoles, 10)
+	validateAssignee(s.T(), usersToBeAsserted, newSpace.SpaceID(), assignedRoles)
+}
+
+func (s *roleManagementServiceBlackboxTest) TestAssignRoleAlreadyExists() {
+	g := s.DBTestSuite.NewTestGraph()
+	newSpace := g.CreateSpace(g.ID("myspace"))
+
+	userToBeAssigned := g.CreateUser(g.ID("somename"))
+	err := s.repo.Assign(context.Background(), userToBeAssigned.Identity().ID, newSpace.SpaceID(), authorization.AdminRole)
+	require.Nil(s.T(), err)
+
+	err = s.repo.Assign(context.Background(), userToBeAssigned.Identity().ID, newSpace.SpaceID(), authorization.AdminRole)
+	require.IsType(s.T(), errors.DataConflictError{}, errs.Cause(err))
+}
+
+func (s *roleManagementServiceBlackboxTest) TestAssignRoleResourceNotFound() {
+	err := s.repo.Assign(context.Background(), uuid.NewV4(), uuid.NewV4().String(), uuid.NewV4().String())
```

Comment:
This will fail because the identity is not found. Not because the resource is not found.
Use some existing identity and existing role in this test.

Reply:
Not really, the check for resource's existence is done first https://github.com/fabric8-services/fabric8-auth/pull/481/files/92b41bdb74998622b4504406c8b41e3f920e5500#diff-98f55d2fa7b5423f02129bc536e43b09R52

Reply:
Added it - it does make sense from an API contract perspective to have *all but one* valid inputs 56ce11be50f5a3f4a519dbfa0343a562c392c878